### ISSUE TITLE
"Save-as" does not work

### ISSUE THREAD

**Issue by dcfreire** [typora/typora-issues#5275]
- [x] Searched existing issues to avoid creating duplicates.

- [x] Confirmed that it can be reproduced in built-in themes without customized css.

   If it only exists in 3rd party themes or css, you can still report it, but please attach the theme target or the css file. We may not "fix" it, if it is caused by 3rd party themes or css styles that we cannot support.

- [x] Searched http://support.typora.io/

----

**Describe the bug**
Ever since release (ver >= 1.0) I can't use the save-as functionality for any files or export files. This is a big problem considering that I cannot create new files on Typora, and then open it on Typora to edit it. When trying clicking save-as or the export buttons nothing happens on screen, but this is printed to the terminal:
```
(typora:114856): Gtk-WARNING **: 15:47:01.744: Can't open portal file chooser: GDBus.Error:org.freedesktop.portal.Error.InvalidArgument: invalid filter: no filters
```

The file chooser opens normally when opening files and folders.

**To Reproduce**
1. Go to File
2. Click on Save-as

**Expected behavior**
The file chooser opens, and lets me save the file.

**Screenshots / Screencasts**
If applicable, add screenshots / screencasts to help explain your problem.

**Desktop (please complete the following information):**
 - OS: Manjaro 21.2.6

**Typora Version**
1.2.4

**Additional context**
Save-as works fine on typora 0.11

**Comment by abnerlee** [comment]
Does `GTK_USE_PORTAL=0 ./Typora` work?

Or install `xdg-desktop-portal-gtk`

**Comment by funnisquares** [comment]
I already had `xdg-desktop-portal-gtk` installed and it does not fixes the issue, however setting `GTK_USE_PORTAL=0` does.

**Comment by davidfhill** [comment]
I tested two Typora versions on Linux Mint 20.03 Una.
Version 1.3.6 (installed through the Linux Mint 20.03 Software Manager) does not open the save/save as dialog when clicking Save or Save As.
Version  1.2.3 (available for download from the typora.io website) functions properly as expected when clicking Save or Save As.
The bug appears to have been introduced in version 1.3.

**Comment by plumlis** [comment]
Same here.
But this issue is only on Flatpak version.
It's normal with deb and binary.

Fedora 36 64 bit 
Gnome - wayland

**Comment by sirrahn** [comment]
Am also experiencing this with the flatpak version on Fedora 36, gnome wayland and gnome xorg.

**Comment by sirrahn** [comment]
How do you set the GTK_USE_PORTAL=0 ?

**Comment by exergonic** [comment]
> How do you set the GTK_USE_PORTAL=0 ?

From a command line, invoke `typora` but prefix it with the desired environmental variable, i.e.
`GTK_USE_PORTAL=0 typora`

**Comment by prototact** [comment]
I tried with `GTK_USE_PORTAL=0`. Whenever I try to "Save As" or "Export" it logs the following

(Typora:2): Gtk-WARNING **: 18:21:09.938: Can't open portal file chooser: GDBus.Error:org.freedesktop.portal.Error.InvalidArgument: invalid filter: no filters

**Comment by benji88i** [comment]
Hi there,

So for us fedora users who don't know how to run the unpack and run the binary version and are stuck with the issues with flatpak (still unable to print or save files), is there a way to build an rpm package or is there a copr repo or rpm version that is trusted?

Maybe the snap version would be better supported..

**Comment by plumlis** [comment]
> Hi there,
> 
> So for us fedora users who don't know how to run the unpack and run the binary version and are stuck with the issues with flatpak (still unable to print or save files), is there a way to build an rpm package or is there a copr repo or rpm version that is trusted?
> 
> Maybe the snap version would be better supported..

https://github.com/RPM-Outpost/typora

try this script

**Comment by benji88i** [comment]
Thanks a  lot!

Sorry do I need to download something from github before I run "./create-package.sh"

**Comment by benji88i** [comment]
Is there anyway to rollback the flathub version. ?

**Comment by akosma** [comment]
Same issue here; Typora 1.3.6 on Ubuntu 20.04, installed with Flatpak. `xdg-desktop-portal-gtk` installed.
Setting `flatpak override --user --env=GTK_USE_PORTAL=0 io.typora.Typora` does not solve the issue.

**Comment by bluesangel** [comment]
Issue persists on FedoraCore 36 after setting GTK_USE_PORTAL=0 through Flatseal.
This environment variable is not a working fix.

**Comment by abnerlee** [comment]
Does version 1.4.4 works?

**Comment by plumlis** [comment]
Still doesn't work.

(Typora:2): Gtk-WARNING **: 15:34:20.227: Can't open portal file chooser: GDBus.Error:org.freedesktop.portal.Error.InvalidArgument: invalid filter: no filters

**Comment by sonnie-sonnig** [comment]
> Does version 1.4.4 works?

Yes for the manual .deb-File-Update. 
No for the Flatpak.
(PopOS 22.04 / Gnome 42.3.1 / X11)

**Comment by abnerlee** [comment]
Could you try 1.4.5 (flatpak)? It should be fixed in that version

**Comment by sonnie-sonnig** [comment]
> Could you try 1.4.5 (flatpak)? It should be fixed in that version

Unfortunately not.
This is my terminal output, when starting Typora with `flatpak run io.typora.Typora`
```

Gtk-Message: 08:35:28.778: Failed to load module "appmenu-gtk-module"
Gtk-Message: 08:35:28.814: Failed to load module "canberra-gtk-module"
Gtk-Message: 08:35:28.814: Failed to load module "canberra-gtk-module"
[2:0908/083528.825389:ERROR:bus.cc(398)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0908/083528.825417:ERROR:bus.cc(398)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0908/083528.860230:ERROR:bus.cc(398)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0908/083528.860281:ERROR:bus.cc(398)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:0908/083528.860307:ERROR:bus.cc(398)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory

(Typora:2): Gtk-WARNING **: 08:35:33.572: Can't open portal file chooser: GDBus.Error:org.freedesktop.portal.Error.InvalidArgument: invalid filter: no filters
```

**Comment by abnerlee** [comment]
@sonnie-sonnig Sorry, but does Export → PDF now shows the dialog in 1.4.5?

**Comment by sonnie-sonnig** [comment]
@abnerlee Yes, this seems to work properly with the flatpak 1.4.5.

**Comment by abnerlee** [comment]
@sonnie-sonnig Could you try 1.4.6 and see if it works now?

**Comment by plumlis** [comment]
It works well now.

**Comment by abnerlee** [comment]
Fixed in 1.4.6

**Comment by sonnie-sonnig** [comment]
> It works well now.

I second that. Works fine with 1.4.6 from flathub.
TY for your work.
